### ISSUE TITLE
New TFX IR MLMD does not store pipeline_name property in pipeline_run context

### ISSUE THREAD

**Issue by casassg** [tensorflow/tfx#3281]
We have internal code that relies on using Context properties in MLMD for retrieving all runs related to a pipeline. Seems new TFX IR does not store property to retrieve pipeline name from pipeline_run context.

Related internal code: #2415 `PipelineContext -> _get_runs`

**Comment by 1025KB** [comment]
It should be part of the [PipelineInfo](https://github.com/tensorflow/tfx/blob/master/tfx/orchestration/data_types.py#L103)

**Comment by ymyfish** [comment]
Hi there, the new IR publisher does publish node context as part of the context of an execution.
https://github.com/tensorflow/tfx/blob/master/tfx/orchestration/portable/execution_publish_utils.py#L81

Could you give me a link to where you read the MLMD context to extract the pipeline_name? Let me double check if the data structure is changed.

**Comment by casassg** [comment]
@ymyfish https://github.com/casassg/tfx/blob/gcasassaez/mlmd_client/tfx/experimental/metadata_client/client.py#L139

I read all context of `pipeline_run` type to extract what contexts are for a specific pipeline. If there's a better alternative it would be great.

**Comment by casassg** [comment]
Repro steps:
- Run a Pipeline w new BeamDagRunner
- Try to extract pipeline runs with the mentioned code (pipeline runs will be empty even after a successful run with BeamDagRunner)

**Comment by ruoyu90** [comment]
@casassg yeah we did not populate those properties in the new IR stack. However you can get exactly the same info by accessing the context name. The pipeline context will have the pipeline name as its name while the pipeline run context will have the pipeline run id as its name.

**Comment by casassg** [comment]
How can I get all pipeline_run context of a pipeline then? Contexts don't  have direct relationships between them which makes it basically impossible to pull all pipeline_run context for a specific pipeline. 

Only option I can think of is: 
- Get all artifacts for a pipeline context
- Get contexts for those artifacts
- Filter contexts by type

**Comment by ymyfish** [comment]
I tried an internal change:

I assume this is what you currently have in the old stack:

```
pipeline_run_contexts = [
      c for c in metadata_connection.store.get_contexts_by_type(metadata._CONTEXT_TYPE_PIPELINE_RUN) 
      if c.properties['pipeline_name'].string_value == pipeline_name
 ]
```

And the following code in the new IR stack gets you the same pipeline_run_context:

```
  pipeline_run_contexts = [
      c for c in metadata_connection.store.get_contexts()
      if c.name == pipeline_name
  ]
```

**Comment by jiyongjung0** [comment]
FWIW, the context returned from the above code by(@ymyfish) is 'pipeline_contexts' not 'pipeline_run_contexts'. I think that the idea from @casassg is the valid approach if pipeline runs are needed.

**Comment by casassg** [comment]
That makes more sense. I wonder though if we can instead add properties? that should allow us to have a weak link between pipeline_run context and pipeline context

**Comment by casassg** [comment]
This could be addressed by using the new ParentContext feature in MLMD 1.X (aka Pipeline is a ParentContext for PipelineRun context)
